client/src/stores/cartStore.ts

```diff
@@ -0,0 +1,88 @@
+import { makeAutoObservable, observable } from 'mobx';
+import CartItem from '../models/cartItem';
+import { isNone, isNotNone } from '../utils/typeGuard';
+
+class CartStore {
+  @observable
+  cartItemList: CartItem[] = [];
+  modalCartItemId: number;
+
+  constructor() {
+    makeAutoObservable(this);
+
+    const cartItemList = JSON.parse(localStorage.getItem('cart') as string);
+    this.cartItemList = cartItemList.map((cart: CartItem) => new CartItem(cart));
+    this.modalCartItemId = 0;
+  }
+
+  addProductToCart(productId: number, title: string, imgSrc: string, count: number, price: number) {
+    const prevItem = this.cartItemList.find((cartItem) => cartItem.id === productId);
+    if (isNotNone(prevItem)) {
+      // 이미 장바구니에 추가된 상품.
+      return;
+    }
+
+    const newCartItem = new CartItem({
+      id: productId,
+      title,
+      imgSrc,
+      count,
+      price,
+    } as CartItem);
+    this.cartItemList.push(newCartItem);
+    this.setCartItemList(this.cartItemList);
+  }
+
+  getModalCartItem() {
+    return this.getCartItemById(this.modalCartItemId);
+  }
+
+  setModalCartItemCount(count: number) {
+    if (isNone(count) || Number.isNaN(count) || count <= 0) {
+      count = 0;
+    }
+    const modalCartItem = this.getCartItemById(this.modalCartItemId);
+    modalCartItem.count = isNotNone(count) ? count : modalCartItem.count;
+    this.cartItemList = [...this.cartItemList];
+    this.setCartItemList(this.cartItemList);
+  }
+
+  setCartItemSelection(id: number, isSelected: boolean) {
+    const cartItem = this.getCartItemById(id);
+    cartItem.isSelected = isSelected;
+    this.cartItemList = [...this.cartItemList];
+    this.setCartItemList(this.cartItemList);
+  }
+
+  setCartItemSelectionAll(isSelected: boolean) {
+    for (const cartItem of this.cartItemList) {
+      cartItem.isSelected = isSelected;
+    }
+
+    this.cartItemList = [...this.cartItemList];
+    this.setCartItemList(this.cartItemList);
+  }
```

Comment:
```suggestion
  setCartItemSelectionAll(isSelected: boolean) {
    this.cartItemList = this.cartItemList.map((cartItem) => {
      cartItem.isSelected = isSelected;
      return cartItem;
    });
    this.setCartItemList(this.cartItemList);
  }
```

이런 식으로 고차함수도 활용해서 할 수 있을 것 같습니다 :)